.travis_scripts/maybe-start-docker.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+# Starts our build image inside docker, if we're doing a docker build.
+
+set -ex
+
+if [ -z "${IMAGE}" ] ; then
+    exit 0
+fi
+
+docker build -t node-rust:"${IMAGE}" -f ".travis_scripts/${IMAGE}/Dockerfile" .
+docker run --detach --name target -v "$(pwd)":/src -w /src -e NPM_TOKEN="$NPM_TOKEN" node-rust:"${IMAGE}" sleep 999999999
```

Comment:
Maybe comment this a bit more to explain what it's doing? I don't immediately understand the `sleep` call.

Reply:
It keeps the container around so it doesn't immediately exit when started detached (since it's not running a process, just sitting there waiting for commands). That's about the extent of what I know about it, came from the ironoxide-java one that @cjyar made.

Reply:
Ok, that's enough of an explanation for me, can we add a comment to here with that?